File: iasa-cp3/src/lib/ecr/Reaccao.py
from comportamento import Comportamento

class Reaccao(Comportamento):

    def activar(self, percepcao):
        estimulo = self.__detetar_estimulo(percepcao)
        if estimulo is not None and estimulo is not False:
            resposta = self.__gerar_resposta(estimulo)
            return resposta

    def __gerar_resposta(self, estimulo):
        raise NotImplementedError

    def __detetar_estimulo(self, percepcao):
        raise NotImplementedError

File: IASA/DISCUSSÃO/iasa-cp1/src/Reaccao/Reaccao.java
package Reaccao;

public class Reaccao implements Comportamento {
	
	private Estimulo estimulo;
	private Accao accao;
	
	public Reaccao(Estimulo estimulo, Accao accao) {
		this.estimulo = estimulo;
		this.accao = accao;
	}

	@Override
	public Accao ativar(Estimulo estimulo) {
		return estimulo == this.estimulo ? accao : null;
	}
}
